### ISSUE TITLE
std::experimental::filesystem is deprecated for recent MSVC versions causing compilation failures

### ISSUE THREAD

**Issue by Q-Minh** [eyalroz/cuda-api-wrappers#343]
The `jitify` example program complains about std::experimental::filesystem being deprecated by Microsoft when using C++17 with VS2019. Assuming the `__cplusplus` is correctly set in the build process, I believe it should suffice to have something like:
```
#if __cplusplus >= 201703 && defined(_MSC_VER)
  #include <filesystem>
  using fs = std::filesystem;  
#else
  #include <experimental/filesystem>
  using fs = std::experimental::filesystem;
#endif
```

**Comment by eyalroz** [comment]
Ok, let's hope this does the trick.

**Comment by Q-Minh** [comment]
Okay, so I just noticed that in the CMakeLists.txt for the examples, you also link `jitify` with `stdc++fs`, which isn't available for VS2019 with C++17. After having thought about it, I think it's fine to keep it that way, because you are the one managing how you want to build the examples locally. Users of your library get to decide if they want to build in C++17 with MSVC for their own CMake targets, and they will specify if they need to link to stdc++fs or not, so it's fine to keep your examples c++11/14 compatible. 
If you do want your examples to be C++17 compatible for VS2019, then I would simply add some CMake option similar to the `-DBUILD_EXAMPLES=ON`, you can add some `-DBUILD_EXAMPLES_WITH_STANDARD=17` for example. But that's up to you. 
Also, it'd be nice if you could prefix your CMake options with some representative name for your library, for example `-DCUDA_API_WRAPPERS_BUILD_EXAMPLES=ON`, so that the option name doesn't clash with any other dependency that users may have.

**Comment by eyalroz** [comment]
> Also, it'd be nice if you could prefix your CMake options with some representative name for your library, for example -DCUDA_API_WRAPPERS_BUILD_EXAMPLES=ON, so that the option name doesn't clash with any other dependency that users may have.

Separate bug for that :-)

**Comment by eyalroz** [comment]
>  you also link jitify with stdc++fs, which isn't available for VS2019 with C++17

IIANM. that's supposed to do nothing for when the fs library isn't separate. But I may be wrong.

**Comment by Q-Minh** [comment]
> > you also link jitify with stdc++fs, which isn't available for VS2019 with C++17
> 
> IIANM. that's supposed to do nothing for when the fs library isn't separate. But I may be wrong.

Yeah, compilation will still succeed, you'll just get an annoying warning from MSVC.

**Comment by eyalroz** [comment]
So... I can close this?

**Comment by Q-Minh** [comment]
Yes.
